olwebsite-app/src/components/Carousel/Carousel.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useCallback, useState } from "react";
+import { FaChevronLeft, FaChevronRight } from "react-icons/fa";
+import PropTypes from "prop-types";
+
+import DotIndicator from "../DotIndicator/DotIndicator";
+
+/************************************
+ * Constants
+ ************************************/
+
+const SLIDE_SIZE = 100;
+
+export default function Carousel ({
+  slides,
+  interval,
+  arrowSize,
+  arrowColor,
+  isAutomatic, 
+  initialPosition,
+  includeDotIndicators,
+  includeNavigationArrows 
+}) {
+
+  /************************************
+   * State
+   ************************************/
+  const [autoRun, setAutoRun] = useState(isAutomatic);
+  const [position, setPosition] = useState(initialPosition);
+
+  /************************************
+   * Callbacks
+   ************************************/
+
+  const handleNext = useCallback(() => {
+    const lastSlide = -SLIDE_SIZE * (slides.length - 1);
+
+    position ===  lastSlide
+      ? setPosition(initialPosition)
+      : setPosition(position - SLIDE_SIZE);
+  }, [initialPosition, position, slides.length]); 
+
+  /************************************
+   * Life Cycle Hooks
+   ************************************/
+
+  useEffect(() => {
+    if(autoRun) {
+      const id = setInterval(handleNext, interval);
+      return () => clearInterval(id);
+    }
+  }, [handleNext, interval, autoRun]);
+ 
+  /************************************
+   * Functions
+   ************************************/
+
+  function handlePrevious() {
+    const lastSlide = -SLIDE_SIZE * (slides.length - 1);
```

Comment:
could move the lastSlide calculation to a utility function as it's used multiple times